src/jit/codegenlegacy.cpp

```diff
@@ -11364,8 +11364,6 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
             if (VarSetOps::IsMember(compiler, genUpdateLiveSetForward(op2), varDsc->lvVarIndex))
             {
-                noway_assert(compiler->opts.compDbgCode);
```

Comment:
Note that this case will arise if we are assigning to a lclVar that is currently live, which shouldn't happen except in the case where we are artificially keeping lclVars live everywhere, which doesn't seem to be the case here. Have you checked that the liveness is computed correctly in this case?